### PR TITLE
Feature/bufr ext

### PR DESCRIPTION
**Pull request by rmclaren**
## Description

- Adds support for transforms to BUFR exports.
- Fixes bug were in memory (ObsSpace backend) creation of numerical value would fail because of a type mismatch (used floats instead of doubles)
- Fixed bug were in memory (ObsSpace backend) creation of std::string variables would fail (worked around ioda-engines bug (zenhub::ioda-engines:140 )).
- Changed name of main executable to toIoda.x to accommodate the fact that it might do more than just BUFR.
- Added backend parameter to YAML file so that the user can explicitly set either "inmemory" or "netcdf" backends.

## Dependencies

- waiting on JCSDA/ioda-converter/pull/331

### CODE REVIEW

**Review by aerorahul — APPROVED** [review]
I reviewed these changes with respect to PR #331 and they are good.

**Comment by aerorahul** [comment]
when #331 is merged into `develop`,  please rebase so we don't see all the same commits in the merge log.

File: src/bufr/BufrParser/Exports/Transforms/OffsetTransform.cpp [hunk 1]
```diff
@@ -0,0 +1,20 @@
+//
+// Created by Ronald McLaren on 9/24/20.
```
Comment (rhoneyager) [inline src/bufr/BufrParser/Exports/Transforms/OffsetTransform.cpp:2]: Copyright statement needed.

File: src/bufr/BufrParser/Exports/Transforms/OffsetTransform.h [hunk 2]
```diff
@@ -0,0 +1,23 @@
+//
+// Created by Ronald McLaren on 9/24/20.
```
Comment (rhoneyager) [inline src/bufr/BufrParser/Exports/Transforms/OffsetTransform.h:2]: Same

**Review by rhoneyager — COMMENTED** [review]
Copyright statements need to follow standards.

File: src/bufr/CMakeLists.txt [hunk 3]
```diff
@@ -60,8 +67,8 @@ add_library( ${PROJECT_NAME}::ingester ALIAS ingester )
 add_executable(test_ingester.x test_ingester.cpp)
 target_link_libraries(test_ingester.x PRIVATE ${PROJECT_NAME}::ingester)
 
-add_executable(convertBufrToIoda.x main.cpp)
-target_link_libraries(convertBufrToIoda.x PRIVATE ${PROJECT_NAME}::ingester)
+add_executable(toIoda.x main.cpp)
+target_link_libraries(toIoda.x PRIVATE ${PROJECT_NAME}::ingester)
```
Comment (CoryMartin-NOAA) [inline src/bufr/CMakeLists.txt:71]: toIoda.x is not explicit enough IMO. Can we at least make it Convert2Ioda.x or something like that?

File: src/bufr/IodaEncoder/IodaDescription.cpp [hunk 4]
```diff
@@ -89,4 +111,20 @@ namespace Ingester
     {
         variables_.push_back(variable);
     }
+
+    void IodaDescription::setBackend(std::string backend)
+    {
+        if (backend == "netcdf")
```
Comment (CoryMartin-NOAA) [inline src/bufr/IodaEncoder/IodaDescription.cpp:117]: if the code supports "inmemory" and "in-memory", can we throw in here "netCDF" and "NetCDF" just for safety's sake?

Comment (CoryMartin-NOAA) [reply src/bufr/IodaEncoder/IodaDescription.cpp:117]: Maybe any case variation of netCDF?

File: src/bufr/BufrParser/Exports/Transforms/OffsetTransform.h (hunk 2 above)
Comment (rmclaren) [reply src/bufr/BufrParser/Exports/Transforms/OffsetTransform.h:2]: OBE (Overcome by Events) in a later branch.

File: src/bufr/CMakeLists.txt (hunk 3 above)
Comment (rmclaren) [reply src/bufr/CMakeLists.txt:71]: updated to convertToIoda.x

File: src/bufr/IodaEncoder/IodaDescription.cpp (hunk 4 above)
Comment (rmclaren) [reply src/bufr/IodaEncoder/IodaDescription.cpp]: Backend is now case insensitive.

**Pull request merged**